go/libkb/leveldb_test.go

```diff
@@ -235,14 +243,24 @@ func TestLevelDb(t *testing.T) {
 					if err = tr.Put(key, nil, []byte{41}); err != nil {
 						t.Fatal(err)
 					}
-					if err = tr.Commit(); err != nil {
+
+					// We do some IO here to give Go's runtime a chance to schedule
+					// different routines and channel operations, to *hopefully* make
+					// sure:
+					// 1) The channel operation is done;
+					// 2) If there exists, any broken OpenTransaction() implementation
+					//		that does not block until this transaction finishes, the broken
+					//		OpenTransaction() would have has returned
+					if err = doSomeIO(); err != nil {
 						t.Fatal(err)
 					}
 
-					select {
```

Comment:
The test failure was a flake, where this line happened after [line 261 below in old version](https://github.com/keybase/client/pull/4752/files#diff-1357c23f2f58d11f9d3b1e95c2a423beL261). It could be reproduced by adding a `time.Sleep(time.Second)` right before this line.

This PR changes the channel to a buffered one, and sends the signal into it before actually calling `Commit()`, so that it doesn't cause failure.

To (hopefully) make sure a mis-behaved transaction implementation fails the test, [some I/O](https://github.com/keybase/client/pull/4752/files#diff-1357c23f2f58d11f9d3b1e95c2a423beR254) is done before actually sending the signal to give the imaginary `OpenTransaction()` a chance (without delaying the test) to return before the signal arrives.